data/hooks/encoding.py

```diff
@@ -189,3 +189,118 @@ def update_tracked_features(self, x: BaseDataManager) -> None:
         TODO: implement autodetection of features to encode
         """
         self.tracked_features = shallow_copy(self.explicit_features)
+
+
+@registered_data_hook("ladder_encode")
+class LadderEncoding(FittedDataHook):
+    """
+    Extension of one-hot-encoding which allows for a specified order to be preserved
+    """
```

Comment:
what do you think about adding an example? e.g.
```suggestion
        Input:
            size
            ----
            small
            medium
            large

        Output:
            size_large  size_medium  size_small
            ----------  -----------  ----------
            0           0            1
            0           1            1
            1           1            1
    """
```

note that I changed `"order": ["small", "medium", "large"]` to "order": ["large", "medium", "small"] at the following line to get the output above https://github.com/SomeoneInParticular/modular_optuna_ml/blob/756a823502161d308389b26217c85256537d82b2/testing/iris_data/iris_config.json#L25

With `"order": ["small", "medium", "large"]` I'm getting:

<img width="1439" alt="image" src="https://github.com/user-attachments/assets/1eb86c31-1621-4b8b-a060-04dc60fddce8" />

i.e. `small` is coded as `1 1 1`

Reply:
Sure, I think thats smart. I think doing so for every hook that transforms the data is smart though, and thus this should be in its own branch (and PR). Thoughts?

Reply:
Agree! 👍🏻